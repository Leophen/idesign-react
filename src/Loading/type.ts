export interface LoadingProps {
  /**
   * 类名
   */
  className?: string;
  /**
   * 自定义样式
   */
  style?: React.CSSProperties;
  /**
   * 加载图标
   * @default Loading
   */
  icon?: string;
  /**
   * 加载图标大小
   * @default 36
   */
  size?: number;
  /**
   * 加载图标颜色
   */
  color?: string;
  /**
   * 旋转提示内容
   */
  info?: React.ReactNode;
  /**
   * 旋转内容
   */
  spinner?: React.ReactNode;
}
