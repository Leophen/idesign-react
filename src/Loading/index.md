---
nav:
  title: 组件
  path: /components
group:
  title: 数据展示组件
  order: 4
order: 3
---

# Loading 加载中

用于显示正在加载状态的组件。

## 基本用法

```tsx
import React from 'react';
import { Loading } from 'idesign-react';

const App = () => {
  return (
    <Loading />
  );
};

export default App;
```

## 旋转提示

```tsx
import React from 'react';
import { Loading } from 'idesign-react';

const App = () => {
  return (
    <Loading info="加载中" />
  );
};

export default App;
```

## 自定义旋转图标

可通过 `icon` 属性自定义图标，[点击查看可选图标](./icon#%E5%85%A8%E9%83%A8%E5%9B%BE%E6%A0%87)。

```tsx
import React from 'react';
import { Loading } from 'idesign-react';

const App = () => {
  return (
    <div className="idesign-demo-block-row">
      <Loading icon="RefreshRight" />
      <Loading icon="SettingFill" />
    </div>
  );
};

export default App;
```

## 自定义旋转图标大小

可通过 `size` 属性自定义图标。

```tsx
import React from 'react';
import { Loading } from 'idesign-react';

const App = () => {
  return (
    <div className="idesign-demo-block-row">
      <Loading size={16} />
      <Loading size={32} />
      <Loading size={48} />
    </div>
  );
};

export default App;
```

## 自定义旋转图标颜色

可通过 `color` 属性自定义图标。

```tsx
import React from 'react';
import { Loading } from 'idesign-react';

const App = () => {
  return (
    <div className="idesign-demo-block-row">
      <Loading color="#E06B67" />
      <Loading color="#DBA049" />
      <Loading color="#7EBF50" />
    </div>
  );
};

export default App;
```

## 自定义旋转内容

可通过 `spinner` 属性自定义旋转内容。

```tsx
import React from 'react';
import { Loading } from 'idesign-react';

const App = () => {
  return (
    <div className="idesign-demo-block-row">
      <Loading spinner="🌞" />
      <Loading spinner={<span>※</span>} />
    </div>
  );
};

export default App;
```

<API />
