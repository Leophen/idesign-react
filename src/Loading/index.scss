@keyframes loading-rotate {
  to {
    transform: rotate(360deg);
  }
}

.i-loading {
  display: flex;
  align-items: center;
}

.i-loading-spinner {
  animation: loading-rotate 2s linear infinite;
  .i-icon {
    color: var(--i-primary-txt);
  }
}

.i-loading-info {
  margin-left: 8px;
  color: var(--i-primary-txt);
}
