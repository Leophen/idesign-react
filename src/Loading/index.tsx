import React from 'react';
import classNames from 'classnames';
import './index.scss';
import Icon from '../Icon';
import { LoadingProps } from './type';

const Loading: React.FC<LoadingProps> = (props) => {
  const {
    className,
    style,
    icon = "Loading",
    size = 36,
    color,
    info = '',
    spinner,
    ...restProps
  } = props;

  return (
    <div
      className={classNames(
        'i-loading',
        className
      )}
      style={{ ...style }}
      {...restProps}
    >
      <div className="i-loading-spinner">
        {spinner ||
          <Icon
            name={icon}
            size={size}
            color={color}
          />
        }
      </div>
      <div
        className="i-loading-info"
        style={{ color }}
      >
        {info}
      </div>
    </div>
  );
};

Loading.displayName = 'Loading';

export default Loading;
