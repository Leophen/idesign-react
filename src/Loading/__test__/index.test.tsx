import { render } from '@testing-library/react';
import Loading from '../index';

describe('Loading 组件测试', () => {
  it('create', () => {
    const { asFragment } = render(<Loading />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('info', () => {
    const wrapper = render(<Loading info="加载中" />);
    expect(wrapper.queryByText('加载中')).toBeInTheDocument();
  })

  it('icon', () => {
    const { container } = render(<Loading icon="RefreshRight" />);
    expect(container.querySelector('.icon-RefreshRight')).toBeInTheDocument();
  })

  it('size', () => {
    const { container } = render(<Loading size={16} />);
    expect(container.querySelector('.i-icon')).toHaveStyle('font-size: 16px');
  })

  it('color', () => {
    const { container } = render(<Loading color='#4F73D9' />);
    expect(container.querySelector('.i-icon')).toHaveStyle('color: #4F73D9');
  })

  it('spinner', () => {
    const wrapper = render(<Loading spinner='🌞' />);
    expect(wrapper.queryByText('🌞')).toBeInTheDocument();
  })
});

