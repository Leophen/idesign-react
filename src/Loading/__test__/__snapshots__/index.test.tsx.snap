// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading 组件测试 create 1`] = `
<DocumentFragment>
  <div
    class="i-loading"
  >
    <div
      class="i-loading-spinner"
    >
      <i
        class="i-icon icon-Loading"
        style="font-size: 36px;"
      />
    </div>
    <div
      class="i-loading-info"
    />
  </div>
</DocumentFragment>
`;
